File: Programmers/java/LEVEL_1_문자열_내_마음대로_정렬하기.java
import java.util.*;

class Solution {
    public String[] solution(String[] strings, int n) {
        String[] answer = {};

        Arrays.sort(strings, (x, y) -> {
            if (x.charAt(n) < y.charAt(n)) {
                return -1;
            }

            if (x.charAt(n) == y.charAt(n)) {
                return x.compareTo(y);
            }

            return 1;
        });

        answer = strings;

        return answer;
    }
}

File: Programmers/cpp/LEVEL_1_문자열_내_마음대로_정렬하기.cpp
#include <string>
#include <vector>
#include <algorithm>

using namespace std;

int idx = -1;

static bool compare(string a, string b) {
	if (a[idx] < b[idx]) {
		return true;
	}

	if (a[idx] == b[idx]) {
		if (a < b) {
			return true;
		}

		return false;
	}

	return false;
}

vector<string> solution(vector<string> strings, int n) {
	vector<string> answer;

	idx = n;

	sort(strings.begin(), strings.end(), compare);

	for (string s : strings) {
		answer.push_back(s);
	}

	return answer;
}